### PR TITLE
[qoi] New port

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/qoi/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+vcpkg_from_github(
```
Comment (Adela0814) [inline ports/qoi/portfile.cmake:1]: Please add the comment at the top.
`#header-only library`

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Usage tested pass on `x64-windows`.

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/qoi/portfile.cmake (hunk 1 above)
Comment (xiaozhuai) [reply ports/qoi/portfile.cmake:3]: Done~

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**